mass2/core/misc.py

```diff
@@ -154,6 +171,7 @@ def merge_dicts_ordered_by_keys(dict1: dict[int, Any], dict2: dict[int, Any]) ->
 
 
 def concat_dfs_with_concat_state(df1: pl.DataFrame, df2: pl.DataFrame, concat_state_col: str = "concat_state") -> pl.DataFrame:
+    """Concatenate two DataFrames vertically, adding a column to indicate which DataFrame each row came from."""
```

Comment:
Concatenate two DataFrames vertically, adding a column (concate_state) to indicate which DataFrame each row came from.